shoop/core/cache/impl.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+import threading
+import time
+
+from django.conf import settings
+from django.core.cache import caches
+from django.core.signals import request_finished
+from django.utils.encoding import force_str
+
+DEFAULT_CACHE_DURATIONS = {
+    # Add default durations for various namespaces here (in seconds)
+}
+_versions = threading.local()
+
+
+def _clear_versions_for_request(**kwargs):
+    _versions.__dict__.clear()
+
+
+request_finished.connect(_clear_versions_for_request, dispatch_uid="shoop.core.cache._clear_versions_for_request")
+
+
+def _get_cache_key_namespace(cache_key):
+    """
+    Split the given cache key by the first colon to get a namespace.
+
+    :param cache_key: Cache key string
+    :type cache_key: str
+    :return: Cache namespace string
+    :rtype: str
+    """
+    return force_str(cache_key).split(":", 1)[0]
```

Comment:
On Python 2 the `force_str` has no effect if you split it with unicode string:
```
In [1]: b'foo:bar'.split(u":", 1)
Out[1]: [u'foo', u'bar']
```

This should work for Py2&Py3:
```python
return force_str(cache_key).split(str(":"), 1)[0]
```